stix_shifter_modules/alertflex/stix_translation/json/from_stix_map.json

```diff
@@ -0,0 +1,48 @@
+{
+  "ipv4-addr": {
+    "fields": {
+      "src_ip": ["a.srcIp"],
+      "dst_ip": ["a.dstIp"],
+      "ip": ["a.dstIp", "a.srcIp"]
+    }
+  },
+  "network-traffic": {
+    "fields": {
+      "src_port": ["a.srcPort"],
+      "dst_port": ["a.dstPort"]
+    }
+  },
+  "file": {
+    "fields": {
+      "name": ["a.fileName"],
+      "hashes.SHA-256": ["a.hashSha256"],
+      "hashes.SHA-1": ["a.hashSha1"],
+      "hashes.MD5": ["a.hashMd5"]
+    }
+  },
+  "process": {
+    "fields": {
+      "name": ["a.processName"],
+      "pid": ["a.processId"]
+    }
+  },
+  "user": {
+    "fields": {
+      "name": ["a.userName"]
+    }
+  },
+  "host": {
+    "fields": {
+      "agent": ["a.agentName", "a.srcHostname", "a.dstHostname"]
+    }
+  },
+  "alert" : {
```

Comment:
This will need to be a custom object, so `x-alertflex-alert`

Reply:
Hi, here I just want to clarify the logic of alertflex adapter. I use the stix-shifter standard module for the translation of STIX requests to SQL.
But alertflex can work only with JPQL request so this why I use some strange notation like "a.agentName", "a.srcHostname", "a.dstHostname" that are close to alerts fields from alertflex DB.
But these fields are really src hostname or dst hostname. If in alert you can see IP address information you can be sure it was extracted from network traffic.

please see below a description of alert table from alertflex DB


@Entity
@Table(name = "alert")
@XmlRootElement
@NamedQueries({
    @NamedQuery(name = "Alert.findAll", query = "SELECT a FROM Alert a")
    , @NamedQuery(name = "Alert.findByAlertId", query = "SELECT a FROM Alert a WHERE a.alertId = :alertId")
    , @NamedQuery(name = "Alert.findByAlertUuid", query = "SELECT a FROM Alert a WHERE a.alertUuid = :alertUuid")
    , @NamedQuery(name = "Alert.findByRefId", query = "SELECT a FROM Alert a WHERE a.refId = :refId")
    , @NamedQuery(name = "Alert.findByNodeId", query = "SELECT a FROM Alert a WHERE a.nodeId = :nodeId")
    , @NamedQuery(name = "Alert.findBySensorId", query = "SELECT a FROM Alert a WHERE a.sensorId = :sensorId")
    , @NamedQuery(name = "Alert.findByCategories", query = "SELECT a FROM Alert a WHERE a.categories = :categories")
    , @NamedQuery(name = "Alert.findByDescription", query = "SELECT a FROM Alert a WHERE a.description = :description")
    , @NamedQuery(name = "Alert.findByAlertSeverity", query = "SELECT a FROM Alert a WHERE a.alertSeverity = :alertSeverity")
    , @NamedQuery(name = "Alert.findByAlertSource", query = "SELECT a FROM Alert a WHERE a.alertSource = :alertSource")
    , @NamedQuery(name = "Alert.findByAlertType", query = "SELECT a FROM Alert a WHERE a.alertType = :alertType")
    , @NamedQuery(name = "Alert.findByEventId", query = "SELECT a FROM Alert a WHERE a.eventId = :eventId")
    , @NamedQuery(name = "Alert.findByEventSeverity", query = "SELECT a FROM Alert a WHERE a.eventSeverity = :eventSeverity")
    , @NamedQuery(name = "Alert.findBySrcIp", query = "SELECT a FROM Alert a WHERE a.srcIp = :srcIp")
    , @NamedQuery(name = "Alert.findByDstIp", query = "SELECT a FROM Alert a WHERE a.dstIp = :dstIp")
    , @NamedQuery(name = "Alert.findBySrcHostname", query = "SELECT a FROM Alert a WHERE a.srcHostname = :srcHostname")
    , @NamedQuery(name = "Alert.findByDstHostname", query = "SELECT a FROM Alert a WHERE a.dstHostname = :dstHostname")
    , @NamedQuery(name = "Alert.findBySrcPort", query = "SELECT a FROM Alert a WHERE a.srcPort = :srcPort")
    , @NamedQuery(name = "Alert.findByDstPort", query = "SELECT a FROM Alert a WHERE a.dstPort = :dstPort")
    , @NamedQuery(name = "Alert.findByFileName", query = "SELECT a FROM Alert a WHERE a.fileName = :fileName")
    , @NamedQuery(name = "Alert.findByFilePath", query = "SELECT a FROM Alert a WHERE a.filePath = :filePath")
    , @NamedQuery(name = "Alert.findByHashMd5", query = "SELECT a FROM Alert a WHERE a.hashMd5 = :hashMd5")
    , @NamedQuery(name = "Alert.findByHashSha1", query = "SELECT a FROM Alert a WHERE a.hashSha1 = :hashSha1")
    , @NamedQuery(name = "Alert.findByHashSha256", query = "SELECT a FROM Alert a WHERE a.hashSha256 = :hashSha256")
    , @NamedQuery(name = "Alert.findByProcessId", query = "SELECT a FROM Alert a WHERE a.processId = :processId")
    , @NamedQuery(name = "Alert.findByProcessName", query = "SELECT a FROM Alert a WHERE a.processName = :processName")
    , @NamedQuery(name = "Alert.findByProcessCmdline", query = "SELECT a FROM Alert a WHERE a.processCmdline = :processCmdline")
    , @NamedQuery(name = "Alert.findByProcessPath", query = "SELECT a FROM Alert a WHERE a.processPath = :processPath")
    , @NamedQuery(name = "Alert.findByUrlHostname", query = "SELECT a FROM Alert a WHERE a.urlHostname = :urlHostname")
    , @NamedQuery(name = "Alert.findByUrlPath", query = "SELECT a FROM Alert a WHERE a.urlPath = :urlPath")
    , @NamedQuery(name = "Alert.findByUserName", query = "SELECT a FROM Alert a WHERE a.userName = :userName")
    , @NamedQuery(name = "Alert.findByAgentName", query = "SELECT a FROM Alert a WHERE a.agentName = :agentName")
    , @NamedQuery(name = "Alert.findByContainerId", query = "SELECT a FROM Alert a WHERE a.containerId = :containerId")
    , @NamedQuery(name = "Alert.findByContainerName", query = "SELECT a FROM Alert a WHERE a.containerName = :containerName")
    , @NamedQuery(name = "Alert.findByLocation", query = "SELECT a FROM Alert a WHERE a.location = :location")
    , @NamedQuery(name = "Alert.findByStatus", query = "SELECT a FROM Alert a WHERE a.status = :status")
    , @NamedQuery(name = "Alert.findByAction", query = "SELECT a FROM Alert a WHERE a.action = :action")
    , @NamedQuery(name = "Alert.findByFilter", query = "SELECT a FROM Alert a WHERE a.filter = :filter")
    , @NamedQuery(name = "Alert.findByInfo", query = "SELECT a FROM Alert a WHERE a.info = :info")
    , @NamedQuery(name = "Alert.findByTimeEvent", query = "SELECT a FROM Alert a WHERE a.timeEvent = :timeEvent")
    , @NamedQuery(name = "Alert.findByTimeCollr", query = "SELECT a FROM Alert a WHERE a.timeCollr = :timeCollr")
    , @NamedQuery(name = "Alert.findByTimeCntrl", query = "SELECT a FROM Alert a WHERE a.timeCntrl = :timeCntrl")})
public class Alert implements Serializable {

    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Basic(optional = false)
    @Column(name = "alert_id")
    private Long alertId;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 37)
    @Column(name = "alert_uuid")
    private String alertUuid;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 255)
    @Column(name = "ref_id")
    private String refId;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 255)
    @Column(name = "node_id")
    private String nodeId;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "sensor_id")
    private String sensorId;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "categories")
    private String categories;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "description")
    private String description;
    @Basic(optional = false)
    @NotNull
    @Column(name = "alert_severity")
    private int alertSeverity;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 32)
    @Column(name = "alert_source")
    private String alertSource;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 32)
    @Column(name = "alert_type")
    private String alertType;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "event_id")
    private String eventId;
    @Basic(optional = false)
    @NotNull
    @Column(name = "event_severity")
    private int eventSeverity;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 128)
    @Column(name = "src_ip")
    private String srcIp;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 128)
    @Column(name = "dst_ip")
    private String dstIp;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 128)
    @Column(name = "src_hostname")
    private String srcHostname;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 128)
    @Column(name = "dst_hostname")
    private String dstHostname;
    @Basic(optional = false)
    @NotNull
    @Column(name = "src_port")
    private int srcPort;
    @Basic(optional = false)
    @NotNull
    @Column(name = "dst_port")
    private int dstPort;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "file_name")
    private String fileName;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "file_path")
    private String filePath;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 128)
    @Column(name = "hash_md5")
    private String hashMd5;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 128)
    @Column(name = "hash_sha1")
    private String hashSha1;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "hash_sha256")
    private String hashSha256;
    @Column(name = "process_id")
    private Integer processId;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "process_name")
    private String processName;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "process_cmdline")
    private String processCmdline;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "process_path")
    private String processPath;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "url_hostname")
    private String urlHostname;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "url_path")
    private String urlPath;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "user_name")
    private String userName;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "agent_name")
    private String agentName;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "container_id")
    private String containerId;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "container_name")
    private String containerName;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "location")
    private String location;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 32)
    @Column(name = "status")
    private String status;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 256)
    @Column(name = "action")
    private String action;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "filter")
    private String filter;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 1024)
    @Column(name = "info")
    private String info;
    @Basic(optional = false)
    @NotNull
    @Size(min = 1, max = 512)
    @Column(name = "time_event")
    private String timeEvent;
    @Column(name = "time_collr")
    @Temporal(TemporalType.TIMESTAMP)
    private Date timeCollr;
    @Column(name = "time_cntrl")
    @Temporal(TemporalType.TIMESTAMP)
    private Date timeCntrl;
    @Lob
    @Size(max = 65535)
    @Column(name = "json_event")
    private String jsonEvent;